app/service/planning_svc.py

```diff
@@ -123,3 +128,14 @@ def _reward_fact_relationship(combo_set, combo_link, score):
         if not operation['cleanup']:
             for link in links:
                 link['cleanup'] = None
+
+    async def _explode_private_facts(self, op_id, agent_id):
```

Comment:
let's shorten this function to:
```
    async def _get_agent_facts(self, op_id, agent_id):
        """
        Collect a list of this agent's facts
        """
        for link in await self.data_svc.dao.get('core_chain', criteria=dict(op_id=op_id, host_id=agent_id)):
            facts = await self.data_svc.dao.get('core_fact', criteria=dict(link_id=link['id']))
            return [f['id'] for f in facts]
```
note the changes

Reply:
The return must be outside the for loop, otherwise (since the links may be more than one as in the execution of File Hunter) I risk losing some agent's facts

I only changed "pf_list" to "agent_facts"